src/v3/core/graph.js

```diff
@@ -20,22 +21,72 @@ export type NeighborsOptions = {|
 
 export opaque type GraphJSON = any; // TODO
 
+const BASE_NODE_ADDRESS = "N\0";
+const BASE_EDGE_ADDRESS = "E\0";
+const SEPARATOR = "\0";
+
+function assertNodeAddress(x: NodeAddress) {
+  if (x == null) {
+    throw new Error(String(x));
+  }
+  if (!x.startsWith(BASE_NODE_ADDRESS)) {
+    if (x.startsWith(BASE_EDGE_ADDRESS)) {
+      throw new Error(`Expected NodeAddress, got EdgeAddress: ${x}`);
+    }
+    throw new Error(`Malformed NodeAddress: ${x}`);
+  }
```

Comment:
Should we check that the input ends with a `SEPARATOR`?

Reply:
Done.